python/revng/cli/_commands/fetch_debuginfo/__init__.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from pathlib import Path, PurePath
+from typing import Set
+
+import pefile
+from elftools.common.exceptions import ELFError
+from elftools.elf.elffile import ELFFile
+
+from revng.cli.commands_registry import Command, CommandsRegistry, Options
+from revng.cli.support import log_error
+
+from .common import log, logger
+from .elf import fetch_dwarf, get_elf_dependencies
+from .pe import fetch_pdb, get_pe_dependencies
+
+default_elf_debug_server = "https://debuginfod.elfutils.org/"
+microsoft_symbol_server_url = "http://msdl.microsoft.com/download/symbols/"
+
+
+def log_success(file_path):
+    log(f"Result: {file_path}")
+
+
+def log_fail():
+    log("Result: No debug info found")
+
+
+class FetchDebugInfoCommand(Command):
+    def __init__(self):
+        super().__init__(("model", "fetch-debuginfo"), "Fetch Debugging Information.")
+
+    def register_arguments(self, parser):
+        parser.add_argument("input", help="The input file.")
+        parser.add_argument(
+            "urls",
+            help="List of URLs where to try finding the debug symbols.",
+            nargs="*",
+            default=[],
+        )
+        parser.add_argument(
+            "-d",
+            "--dependencies",
+            help="Fetch debug info for dependencies as well.",
+            type=int,
+            default=0,
+        )
+
+    def run(self, options: Options):
+        args = options.parsed_args
+
+        self.verbose = args.verbose
+        logger.verbose = self.verbose
+
+        the_input_path = Path(args.input)
+        if not the_input_path.exists():
+            log_error("Could not find " + args.input)
+            return 1
+
+        path = str(the_input_path.resolve())
+        with open(args.input, "rb") as f:
```

Comment:
I think `Path` has an `open` method.
You almost never want to convert it to a `str`.